pkg/operator/configobservation/node/observe_latency_profile.go

```diff
@@ -0,0 +1,127 @@
+package node
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	configV1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+var notReadyTolerationSecondsPath = []string{"apiServerArguments", "default-not-ready-toleration-seconds"}
+var unreachableTolerationSecondsPath = []string{"apiServerArguments", "default-unreachable-toleration-seconds"}
+
+// ObserveNotReadyTolerationSeconds observes the value that should be set for default-not-ready-toleration-seconds
+// api server argument on the basis of provided worker latency profile from config node object.
+func ObserveNotReadyTolerationSeconds(genericListers configobserver.Listers, _ events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
```

Comment:
should those observes be also set on an `SNO` (single node) cluster?

Reply:
For single node cluster(s), the behavior applied by these 2 config observer funcs should be quite idempotent.
Even in case, the cluster topology is SNO applying any worker latency profile shouldn't harm the cluster stability in any way. The two different k-apiserver arguments used here: `default-not-ready-toleration-seconds` and `default-unreachable-toleration-seconds` are essentially latency settings acting only when interaction between multiple control plane nodes come into play. Since this is our very first iteration of worker latency profiles, we're following a very straight forward approach and applied updated args in SNO clusters shouldn't cause any issue.